### ISSUE TITLE
Generating new token results in an invalid url

### ISSUE THREAD

**Issue by jkaberg** [clems4ever/authelia#35]
When registring for a new totp token the url thats sent looks like 

```
https://auth.domain.tldundefined?identity_token=<....token....>&redirect=https://sub.domain.tld/
```

Should be 
```
https://auth.domain.tld/totp-register?identity_token=<....token....>&redirect=https://sub.domain.tld/
```

**Comment by clems4ever** [comment]
Hello @jkaberg , I made a fix for the undefined part in #37. I have just merged it but I think it won't fix your problem. 

This undefined value is here because this URL is generated by concatenating the hostname (which is auth.domain.tld in your case) and the content of the header value **x-original-uri** which is set by nginx proxy in the example configuration. 
I guess it is not set in your case so the value is undefined. My fix protects this behaviour by appending an empty string instead of undefined but you should still have your problem as long as **x-original-uri** is not correctly set by the proxy.

**Comment by SerialVelocity** [comment]
I guess I'm misunderstanding something since this project is pretty new to me, but what should that `undefined` represent? From the URL example that @jkaberg posted, it should just be a "register" URL so it shouldn't matter what `x-original-url` is set to, right?

**Comment by clems4ever** [comment]
Hello @SerialVelocity, the "undefined" you see is the outcome of a bug I've just fixed. The value it represents is the path the user must be redirected to to get its TOTP secret generated.

Indeed, in the case of @jkaberg , it should only be **/totp-register** so no need to use x-original-url for that. Actually a few months ago, I served the authentication API behind a subpath of the protected subdomain (it was /authentication/ ) and this variable was used to generate the correct redirection URL based on that config. Today, since authentication is served by a dedicated subdomain, this variable is now useless. It should disappear at the end of the refactoring of the client anyway.
